server/drivers/linux_input.c

```diff
@@ -183,6 +185,7 @@ linuxInput_init (Driver *drvthis)
 					drvthis->name, s);
 			return -1;
 		}
+		p->name = s;
```

Comment:
I believe we should strdup() here. It probably works without, but I think we have no guarantee, that the server keeps the memory s is pointing at around.

Otherwise this PR looks perfect to me.

Reply:
Heh, I actually checked the lifetime of the string returned by config_get_string before doing things like this, the string is valid until config_clear() gets called.

The only caller of config_clear() is do_reload() from server/main.c which first calls drivers_unload_all() so the driver gets closed before the string returned from config_get_string() becomes invalid. IOW no need to strdup.

Reply:
Yes, it works. But the point is: We have an API between server and drivers and ATM this API doesn't guarantee that the server will keep the string around. Of course we can just update the API in this regard. (I kind of would love to - I hate the amount of pointless strdup() we do for this reason.) So the real question is: Should we do this? I kind of understand the point between not mixing server allocated memory and driver allocated memory in the same data structure, so I never dared proposing something like that. But I'm certainly willing to be convinced.

However, actually the whole question isn't that important, because we probably will switch our configuration management soon after the next release to libelektra ...

Reply:
Ok, if this is going to change anyways I'm going to keep this as is. Let me know if I misunderstood and you still want me to add the strdup.